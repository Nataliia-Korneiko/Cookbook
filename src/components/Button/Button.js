/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import s from './Button.module.css';

const Button = ({ type = 'button', label = '', onClick = () => null }) => {
  return (
    <button className={s.button} type={type} onClick={onClick}>
      {label}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
