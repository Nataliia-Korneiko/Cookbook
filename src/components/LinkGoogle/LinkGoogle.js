import React from 'react';
import { ReactComponent as IconGoogle } from '../../assets/icons/icon-google.svg';
import s from './LinkGoogle.module.css';

const LinkGoogle = () => {
  return (
    <a className={s.link__google} href="https://www.google.com.ua/">
      <div className={s.link__wrapper}>
        <IconGoogle width="28" height="28" />
        <span className={s.link__span}>Google</span>
      </div>
    </a>
  );
};

export default LinkGoogle;
