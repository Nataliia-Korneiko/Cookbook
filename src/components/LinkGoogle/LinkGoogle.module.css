.link__google {
  display: block;
  width: 120px;
  height: 44px;
  border: 2px solid var(--color-golden);
  border-radius: 5px;
  font-family: 'Montserrat', sans-serif;
  font-size: 18px;
  font-weight: 500;
  color: var(--color-black);
  background-color: var(--bg-color-white);
  box-shadow: var(--box-shadow-light);
  outline: none;
  transition: 0.8s ease-in-out;
  box-shadow: var(--box-shadow-light);
}

.link__google:hover,
.link__google:focus {
  background-color: var(--bg-color-golden);
  transition: 0.8s ease-in-out;
}

.link__wrapper {
  display: flex;
  padding-left: 8px;
  padding-top: 6px;
}

.link__span {
  padding-left: 6px;
  line-height: 28px;
}
