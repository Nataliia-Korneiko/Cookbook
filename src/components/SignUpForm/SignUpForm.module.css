.wrapper {
  margin: 0 auto;
  padding: 20px 0;
  width: 300px;
  border: none;
  border-radius: 8px;
  background-image: url(../../assets/img/backdrops/backdrop-06.png);
  background-repeat: no-repeat;
  background-position: center;
  box-shadow: var(--box-shadow-light);
}

.form {
  /* padding: 30px 0; */
  display: flex;
  flex-direction: column;
  align-items: center;
}

.label {
  margin-bottom: 20px;
  position: relative;
  width: 270px;
  font-size: 14px;
  font-weight: 400;
  color: var(--color-grey);
}

.label::before {
  content: '* ';
  color: var(--color-orange);
}

.input {
  padding: 0 10px;
  display: block;
  width: 100%;
  height: 44px;
  border: 2px solid var(--color-golden);
  border-radius: 5px;
  font-size: 16px;
  color: var(--color-dark-blue);
  outline: none;
  box-shadow: var(--box-shadow-light);
}

/* ::placeholder {
  font-size: 13px;
  font-weight: 400;
  color: #a6abb9;
} */

.error {
  position: absolute;
  left: 0;
  bottom: -18px;
  display: block;
  color: var(--color-orange);
  width: 100%;
  font-weight: 500;
  z-index: 1;
}

.error::first-letter {
  text-transform: uppercase;
}

.relink {
  color: var(--color-grey);
}

.link {
  color: var(--color-orange);
}
