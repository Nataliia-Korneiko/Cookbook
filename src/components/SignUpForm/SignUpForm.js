import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import shortid from 'shortid';
import { validateAll } from 'indicative/validator';
import routes from '../../routes/routes';
import LinkGoogle from '../LinkGoogle/LinkGoogle';
import Button from '../ButtonAuthorization/ButtonAuthorization';
import s from './SignUpForm.module.css';

const validationRules = {
  name: 'required|min:2|max:20',
  email: 'required|email',
  password: 'required|min:6|max:12',
};

const validationMessages = {
  'name.required': 'This field is required!',
  'name.min': 'Name must be at least 2 characters!',
  'name.max': 'Name must be no more than 20 characters!',
  'email.required': 'This field is required!',
  'email.email': 'Enter a valid email address!',
  'password.required': 'This field is required!',
  'password.min': 'Password must be at least 6 characters!',
  'password.max': 'Password must be no more than 12 characters!',
};

class SignUpForm extends Component {
  state = {
    name: '',
    email: '',
    password: '',
    error: null,
  };

  ids = {
    nameId: shortid.generate(),
    emailId: shortid.generate(),
    passwordId: shortid.generate(),
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  handleSubmit = e => {
    e.preventDefault();

    const { name, email, password } = this.state;

    validateAll({ name, email, password }, validationRules, validationMessages)
      .then(() => {
        this.setState({ name: '', email: '', password: '', error: null });
      })
      .catch(errors => {
        const formatedErrors = {};
        errors.forEach(error => {
          formatedErrors[error.field] = error.message;
        });
        this.setState({
          error: formatedErrors,
        });
      });
  };

  render() {
    const { name, email, password, error } = this.state;
    return (
      <div className={s.wrapper}>
        <LinkGoogle />
        <form
          className={s.form}
          onSubmit={this.handleSubmit}
          autoComplete="off"
          noValidate
        >
          <label
            className={`${s.label} ${s.label__name}`}
            htmlFor={this.ids.nameId}
          >
            Name&#58;
            <input
              className={s.input}
              id={this.ids.nameId}
              type="text"
              name="name"
              value={name}
              onChange={this.handleChange}
              placeholder="Enter your name"
            />
            {error && <span className={s.error}>{error.name}</span>}
          </label>
          <label
            className={`${s.label} ${s.label__email}`}
            htmlFor={this.ids.emailId}
          >
            E-mail&#58;
            <input
              className={s.input}
              id={this.ids.emailId}
              type="email"
              name="email"
              value={email}
              onChange={this.handleChange}
              placeholder="Enter your email"
            />
            {error && <span className={s.error}>{error.email}</span>}
          </label>
          <label
            className={`${s.label} ${s.label__password}`}
            htmlFor={this.ids.passwordId}
          >
            Password&#58;
            <input
              className={s.input}
              id={this.ids.passwordId}
              type="password"
              name="password"
              value={password}
              onChange={this.handleChange}
              placeholder="Enter your password"
            />
            {error && <span className={s.error}>{error.password}</span>}
          </label>
          <Button label="Sign Up" type="submit" />
          <p className={s.relink}>Already have an account?</p>
          <NavLink to={routes.LOG_IN} className={s.link}>
            Log In
          </NavLink>
        </form>
      </div>
    );
  }
}

export default SignUpForm;
