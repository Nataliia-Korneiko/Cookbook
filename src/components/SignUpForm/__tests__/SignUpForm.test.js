import React from 'react';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';
import SignUpForm from '../SignUpForm';
import Button from '../../ButtonAuthorization/ButtonAuthorization';
import LinkGoogle from '../../LinkGoogle/LinkGoogle';

describe('SignUpForm component', () => {
  const initialState = {
    name: '',
    email: '',
    password: '',
    error: null,
  };

  const newSignUpForm = shallow(<SignUpForm />);

  it('initialize initialState with initial state', () => {
    expect(newSignUpForm.state()).toEqual(initialState);
  });

  it('renders a properly', () => {
    expect(newSignUpForm).toMatchSnapshot();
  });

  it('renders a Button', () => {
    expect(newSignUpForm.find(Button)).toHaveLength(1);
  });

  it('renders a LinkGoogle', () => {
    expect(newSignUpForm.find(LinkGoogle)).toHaveLength(1);
  });

  it('renders one NavLink', () => {
    expect(newSignUpForm.find(NavLink)).toHaveLength(1);
  });

  it('renders two div', () => {
    expect(newSignUpForm.find('div')).toHaveLength(2);
  });

  it('renders one form', () => {
    expect(newSignUpForm.find('form')).toHaveLength(1);
  });

  it('renders three label', () => {
    expect(newSignUpForm.find('label')).toHaveLength(3);
  });

  it('renders one input', () => {
    expect(newSignUpForm.find('input[type="text"]')).toHaveLength(1);
  });

  it('renders one input', () => {
    expect(newSignUpForm.find('input[type="email"]')).toHaveLength(1);
  });

  it('renders one input', () => {
    expect(newSignUpForm.find('input[type="password"]')).toHaveLength(1);
  });

  it('renders one p', () => {
    expect(newSignUpForm.find('p')).toHaveLength(1);
  });

  describe('testing register', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<SignUpForm />);
    });

    it('name check', () => {
      wrapper.find('input[type="text"]').simulate('change', {
        target: { name: 'name', value: 'testname' },
      });

      expect(wrapper.state('name')).toEqual('testname');
    });

    it('email check', () => {
      wrapper.find('input[type="email"]').simulate('change', {
        target: { name: 'email', value: 'test@.ukr.net' },
      });

      expect(wrapper.state('email')).toEqual('test@.ukr.net');
    });

    it('password check', () => {
      wrapper.find('input[type="password"]').simulate('change', {
        target: { name: 'password', value: '123456' },
      });

      expect(wrapper.state('password')).toEqual('123456');
    });
  });
});
