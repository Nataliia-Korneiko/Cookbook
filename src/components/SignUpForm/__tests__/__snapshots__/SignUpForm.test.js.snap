// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm component renders a properly 1`] = `ShallowWrapper {}`;
